ipol_demo/modules/core/archive/archive.py

```diff
@@ -0,0 +1,567 @@
+        """
+        cursor_db = conn.cursor()
+        cursor_db.execute("SELECT * FROM blobs WHERE id = ?", (id_blob,))
+        tmp = cursor_db.fetchone()
+
+        # get the new path of the blob and thumbnail
+        path_blob = None
+        path_thumb = None
+        if tmp is not None:
+            path_blob, _ = self.get_new_path(settings.blobs_dir, tmp[1], tmp[2])
+            path_thumb, _ = self.get_new_path(settings.blobs_thumbs_dir, tmp[1], "jpeg")
+
+        cursor_db.execute("DELETE FROM blobs WHERE id = ?", (id_blob,))
+
+        # delete the files of this blob
+        try:
+            os.remove(path_blob)
+            os.remove(path_thumb)
+        except Exception:
+            pass
+
+    def mkdir_p(path: str) -> None:
+        """
+        Implement the UNIX shell command "mkdir -p"
+        with given path as parameter.
+        """
+        try:
+            os.makedirs(path)
+        except OSError as exc:
+            if exc.errno == errno.EEXIST and os.path.isdir(path):
+                pass
+            else:
+                raise
+
+    def get_hash_blob(self, path: str) -> str:
+        """
+        Return sha1 hash of given blob
+        """
+        with open(path, "rb") as the_file:
+            return hashlib.sha1(the_file.read()).hexdigest()
+
+    def file_format(self, the_file: str) -> str:
+        """
+        Return format of the file
+        """
+        mime = magic.Magic(mime=True)
+        fileformat = mime.from_file(the_file)
+        fileformat = fileformat.split("/")[0]
+        return fileformat
+
+    #####
+    # adding an experiment to the archive database
+    #####
+    def create_experiment(
+        self,
+        demo_id: int,
+        blobs: str = None,
+        parameters: str = None,
+        execution: str = None,
+    ) -> Result[dict, str]:
+        """
+        Add an experiment to the archive.
+        """
+        # # initialize list of copied files, to delete them in case of exception
+        copied_files_list = []
+        try:
+            conn = lite.connect(settings.database_file)
+            id_experiment = self.update_exp_table(conn, demo_id, parameters, execution)
+            dict_corresp = []
+            dict_corresp = self.update_blob(conn, blobs, copied_files_list)
+            self.update_correspondence_table(conn, id_experiment, dict_corresp)
+            conn.commit()
+            conn.close()
+            return Ok({"experiment_id": id_experiment})
+
+        except Exception as ex:
+            message = "Failure in add_experiment. Error = {}".format(ex)
+            self.logger.exception(message)
+            try:
+                # Execute database rollback
+                conn.rollback()
+                conn.close()
+
+                # Execute deletion of copied files
+                for copied_file in copied_files_list:
+                    os.remove(copied_file)
+
+            except Exception:
+                pass
+
```

Comment:
Missing a return for the error case. Used here https://github.com/ipol-journal/ipolDevel/pull/196/files#diff-67df7a992db8fdc8199e09bd24943d112592476d57ef394361ea29df5057fc60R223